models/login_source.go

```diff
@@ -416,6 +416,18 @@ func LoginViaLDAP(user *User, login, password string, source *LoginSource, autoR
 		return nil, ErrUserNotExist{0, login, 0}
 	}
 
+	// Update User if exist
+	isExist, err := IsUserExist(0, sr.Username)
+	if err != nil {
+		return nil, err
+	} else if isExist {
+		user.LowerName = strings.ToLower(sr.Username)
+		user.Name = sr.Username
+		user.FullName = composeFullName(sr.Name, sr.Surname, sr.Username)
+		user.Email = sr.Mail
+		user.IsAdmin = sr.IsAdmin
```

Comment:
If no admin filter is set in LDAP source this will remove admin rights when set manually